### PR TITLE
Proxies: add checks for base being a contract

### PR DESCRIPTION
**Pull request by sohkai**
Adds more contract checks to the proxies.

Bytecode change:

```
                    CODE DEPOSIT COST    DEPLOYED BYTES     INITIALIZATION BYTES
AppProxyPinned.json Same                 0                  -18
KernelProxy.json    Same                 0                  +126
```

### CODE REVIEW

File: contracts/apps/AppProxyPinned.sol [hunk 1]
```diff
@@ -21,7 +22,7 @@ contract AppProxyPinned is AppProxyBase {
         public // solium-disable-line visibility-first
     {
         setPinnedCode(getAppBase(_appId));
-        require(pinnedCode() != address(0));
+        require(isContract(pinnedCode()));
```
Comment (sohkai) [inline contracts/apps/AppProxyPinned.sol]: Note that strictly, this isn't necessary as our `Kernel` doesn't let an app be set unless it's a contract (https://github.com/aragon/aragonOS/blob/dev/contracts/kernel/Kernel.sol#L177), so this is really more of an assertion (should we make it one?).

This also complicates the tests quite a bit, as you'll see in 961a7d7a81fcff0951225f95e72e8df697b5213d

Comment (izqui) [reply contracts/apps/AppProxyPinned.sol]: I don't think an assertion makes sense here, as a contract that follows the `IKernel` interface may not check whether a base is a contract. A require is fine IMO.

File: test/mocks/KernelSetAppMock.sol [hunk 2]
```diff
@@ -0,0 +1,13 @@
+pragma solidity 0.4.18;
+
+import "../../contracts/kernel/Kernel.sol";
+
+contract KernelSetAppMock is Kernel {
+    function KernelSetAppMock() Kernel(false) public {
+    }
+
+    function setApp(bytes32 _namespace, bytes32 _name, address _app) public returns (bytes32 id) {
```
Comment (izqui) [inline test/mocks/KernelSetAppMock.sol]: We should add a comment explaining that we are bypassing the `isContract` check

**Comment by sohkai** [comment]
Yeah... there is a way right now, by overloading the `isContract()` base... but that won't be an option if we turn it into a library (which we probably will after turning on the optimizer).

**Pull request merged**